core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -140,11 +140,9 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
         .header("Content-Type", MediaType.APPLICATION_OCTET_STREAM)
         .build()
     } catch {
-      case NonFatal(e) =>
-        Response.serverError()
-          .entity(s"Event logs are not available for app: $appId.")
-          .status(Response.Status.SERVICE_UNAVAILABLE)
-          .build()
+      case NonFatal(_) =>
+        UIUtils.buildErrorResponse(Response.Status.SERVICE_UNAVAILABLE,
```

Comment:
If doing the cleanup, it's probably ok to just throw the exception here.

Reply:
This could just `throw ServiceUnavailable`, no?

Reply:
no, we need to set the type, otherwise it returns the response as octet-stream which is causing the issue..

Reply:
But isn't the type being set in the exception now? Seems like the same thing that happens in other cases, where the response is set to json but the exception overrides it when thrown.

But if it doesn't really work, then ok.

Reply:
oh, now I see what you mean, sorry, I didn't get it. I am doing that, thanks.